en: update auto-scaling doc for tidb 5.0 (#1070)
pingcap/docs-tidb-operator: en/enable-tidb-cluster-auto-scaling.md

## `TidbClusterAutoScaler`

> The `TidbClusterAutoScaler` CR object is used to control the behavior of the auto-scaling behavior in the TiDB cluster.

Reviewer: ```suggestion
The `TidbClusterAutoScaler` CR object is used to control the auto-scaling behavior in the TiDB cluster.
```

spec:
  cluster:
    name: auto-scaling-demo
  tikv:
    resources:
      storage_small:
        cpu: 1000m
        memory: 2Gi
        storage: 10Gi
        count: 3
    rules:
      cpu:
        max_threshold: 0.8
        min_threshold: 0.2
        resource_types:
        - storage_small
    scaleInIntervalSeconds: 500
    scaleOutIntervalSeconds: 300
  tidb:
    resources:
      compute_small:
        cpu: 1000m
        memory: 2Gi
        count: 3
    rules:
      cpu:
        max_threshold: 0.8
        min_threshold: 0.2
        resource_types:
        - compute_small
```

> ### Working principles

Reviewer: ```suggestion
### Implementation principles
```

> According to the configuration of the `TidbClusterAutoScaler` CR, TiDB Operator sends requests to PD to query the result of scaling. Based on the result, TiDB Operator makes use of the [heterogeneous cluster](deploy-heterogeneous-tidb-cluster.md) feature to create, update, or delete the heterogeneous TiDB cluster to achieve auto-scaling of the TiDB cluster.

Reviewer: （只配置 TiDB 组件或者只配置 TiKV 组件）is missing.

> According to the configuration of the `TidbClusterAutoScaler` CR, TiDB Operator sends requests to PD to query the result of scaling. Based on the result, TiDB Operator makes use of the [heterogeneous cluster](deploy-heterogeneous-tidb-cluster.md) feature to create, update, or delete the heterogeneous TiDB cluster to achieve auto-scaling of the TiDB cluster.

Author: ```suggestion
According to the configuration of the `TidbClusterAutoScaler` CR, TiDB Operator sends requests to PD to query the result of scaling. Based on the result, TiDB Operator makes use of the [heterogeneous cluster](deploy-heterogeneous-tidb-cluster.md) feature to create, update, or delete the heterogeneous TiDB cluster (only the TiDB component or the TiKV component is configured). In this way, the auto-scaling of the TiDB cluster is achieved.
```

Author: @TomShawn PTAL~

### Related fields

* `spec.cluster`: the TiDB cluster to be elastically scheduled.

    * `name`: the name of the TiDB cluster.
    * `namespace`: the namespace of the TiDB cluster. If not configured, this field is set to the same namespace as the `TidbClusterAutoScaler` CR by default.

* `spec.tikv`: the configuration related to TiKV elastic scheduling.
* `spec.tikv.resources`: the resource types that TiKV can use for elastic scheduling. If not configured, this field is set to the same value as `spec.tikv.requests` in the `TidbCluster` CR corresponding to `spec.cluster`.

    * `cpu`: CPU configuration.
    * `memory`: memory configuration.
    * `storage`: storage configuration.

>     * `count`: the number of resources that the current configuration can use. If not configured, there is no limit.

Reviewer: ```suggestion
    * `count`: the number of resources that the current configuration can use. If this field is not configured, there is no limit on resources.
```

* `spec.tikv.rules`: the rules of TiKV elastic scheduling. Currently only CPU-based rules are supported.

>     * `max_threshold`: If the CPU utilization of all Pods is higher than `max_threshold`, the scaling-out operation is triggered.

Reviewer: ```suggestion
    * `max_threshold`: If the average CPU utilization of all Pods is higher than `max_threshold`, the scaling-out operation is triggered.
```

>     * `min_threshold`: If the CPU utilization of all Pods is lower than `min_threshold`, the scaling-in operation is triggered.

Reviewer: ```suggestion
    * `min_threshold`: If the average CPU utilization of all Pods is lower than `min_threshold`, the scaling-in operation is triggered.
```

    * `resource_types`: the resource types that can be used for CPU-based elastic scheduling. This field corresponds to `key` in `spec.tikv.resources[]`. If not configured, this field is set to all `key`s in `spec.tikv.resources[]` by default.

> * `spec.tikv.scaleInIntervalSeconds`: the minimum interval between this scaling-in operation and the last scaling in/out operation. If not configured, the field is set to `300` by default, which means 300 seconds.

Reviewer: ```suggestion
* `spec.tikv.scaleInIntervalSeconds`: the minimum interval between this scaling-in operation and the last scaling in/out operation. If not configured, the field is set to `500` by default, which means 500 seconds.
```

> * `spec.tikv.scaleInIntervalSeconds`: the minimum interval between this scaling-in operation and the last scaling in/out operation. If not configured, the field is set to `500` by default, which means 500 seconds.

Author: ```suggestion
* `spec.tikv.scaleInIntervalSeconds`: the minimum interval between this scaling-in operation and the last scaling in/out operation. If not configured, the field is set to `500` by default, which means 500 seconds.
* `spec.tikv.scaleOutIntervalSeconds`: the minimum interval between this scaling-out operation and the last scaling in/out operation. If not configured, the field is set to `300` by default, which means 300 seconds.
```

Reviewer 2: Remove `minimum`.

Reviewer: LGTM except for `minimum`

Author: `minimum` removed.

> * `spec.tidb`: the configuration related to TiDB elastic scheduling. Other fields are the same as `spec.tikv`.

Reviewer: < * `spec.tikv.scaleOutIntervalSeconds`：扩容距离上一次伸缩（扩容或者缩容）的冷却时间，如果没有配置，默认设置为 `300`，即 `300s`。> is missing.
